core/controlplane/config/templates/cloud-config-controller

```diff
@@ -2689,6 +2697,105 @@ write_files:
            echo "Finished Loading EFS Persistent Volume"'
 
 {{ end }}
+{{if .Experimental.Kube2IamSupport.Enabled }}
+  - path: /srv/kubernetes/manifests/kube2iam-ds.yaml
+    content: |
+      apiVersion: extensions/v1beta1
+      kind: DaemonSet
+      metadata:
+        name: kube2iam
+        namespace: kube-system
+        labels:
+          app: kube2iam
+        annotations:
+          scheduler.alpha.kubernetes.io/critical-pod: ''
+      spec:
+        template:
+          metadata:
+            labels:
+              name: kube2iam
+          spec:
+            updateStrategy:
+              type: RollingUpdate
+            {{- if .Experimental.Plugins.Rbac.Enabled}}
+            serviceAccountName: kube2iam
+            {{- end}}
+            hostNetwork: true
+            containers:
+              - image: jtblin/kube2iam:latest
+                name: kube2iam
+                args:
+                  - "--app-port=8282"
+                  - "--base-role-arn=arn:aws:iam::{{.AWSAccountID}}:role/"
```

Comment:
Thanks for suggestion. Actually helped me realise that by using `--auto-discover-base-arn` and `--auto-discover-default-role` we can use both, role name and role arn. Apart from this we don't have to inject the `AWSAccountID` in template and `--auto-discover-default-role` also provide the necessary role for kube-resources-autosave as fallback fixing #805.